### PR TITLE
An 4769/me solsniper cnft sales

### PR DESCRIPTION
**Pull request by tarikceric**
cNFT sales for solsniper/magic eden
- first get all sales via `_onchain` models, then decode mint via `bronze_api` models, and join into final table

### CODE REVIEW

File: models/silver/nfts/silver__nft_sales_solsniper_cnft_onchain.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+              datepart: day
+              interval: 2
+      - name: BLOCK_ID
+        description: "{{ doc('block_id') }}"
+        tests:
+          - not_null
+      - name: TX_ID
+        description: "{{ doc('tx_id') }}"
+        tests:
+          - not_null
```
Comment (desmond-hui) [inline models/silver/nfts/silver__nft_sales_solsniper_cnft_onchain.yml:24]: Add the included or custom relationship test to make sure all our onchain events make it to the destination table

File: models/silver/nfts/silver__nft_sales_solsniper_cnft.sql [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        ['a.tx_id','b.mint']
+    ) }} AS nft_sales_solsniper_cnft_id,
+    SYSDATE() AS inserted_timestamp,
+    SYSDATE() AS modified_timestamp
+FROM
+    {{ ref('silver__nft_sales_solsniper_cnft_onchain') }} A
+    LEFT JOIN mint_addresses b
+    ON A.tx_id = b.tx_id
+WHERE
+    b.mint IS NOT NULL
```
Comment (desmond-hui) [inline models/silver/nfts/silver__nft_sales_solsniper_cnft.sql:61]: I think we should include all the sales, even if we don't have the mint. Add a test to check for NULL mints so we are alerted and can triage the pipeline

**Review by desmond-hui — CHANGES_REQUESTED** [review]

File: models/silver/nfts/silver__nft_sales_solsniper_cnft_onchain.yml (hunk 1 above)
Comment (tarikceric) [reply models/silver/nfts/silver__nft_sales_solsniper_cnft_onchain.yml:24]: added relationship test for solsniper and magic_eden_cnft_onchain to decoded_instructions_combined -- or were you thinking a relationship between the `_cnft_onchain` and final `_cnft` table?

File: models/silver/nfts/silver__nft_sales_solsniper_cnft.sql (hunk 2 above)
Comment (tarikceric) [reply models/silver/nfts/silver__nft_sales_solsniper_cnft.sql:61]: removed filter from both models. I think the `not_null` test for the `mint` column in the yml should suffice as a check

File: models/silver/nfts/silver__nft_sales_magic_eden_cnft.sql
Comment (desmond-hui) [inline models/silver/nfts/silver__nft_sales_magic_eden_cnft.sql:1]: How would this (and the solsniper model) "auto-heal"? Meaning, at this moment we now have N records w/ NULL mints because we are waiting for a backfill of the bronze data. How do we fix the NULL records as the bronze data arrives?

**Review by desmond-hui — CHANGES_REQUESTED** [review]

File: models/silver/nfts/silver__nft_sales_magic_eden_cnft.sql
Comment (tarikceric) [reply models/silver/nfts/silver__nft_sales_magic_eden_cnft.sql:1]: great point -- added temp backfill logic using an _inserted_timestamp from bronze table, using a temp `mint_inserted_timestamp` column to track the latest records that have went through the parser

**Review by desmond-hui — APPROVED** [review]
